path_2024.py

```diff
@@ -81,7 +82,8 @@ def run() -> int:
         print("Error: upload_commands")
         return -1
 
-    while True:
+    start_time = time.time()
+    while time.time() - start_time < STOP_CONDITION_TIMEOUT:
```

Comment:
Where is the code for the drone completing its current lap and returning to launch point?